src/ios/vfr/SDVReaderViewController.m

```diff
@@ -136,10 +137,56 @@ -(void)updateArticleView
 {
     if ([mainPagebar isKindOfClass:[SDVReaderMainPagebar class]]) {
         SDVReaderMainPagebar* sdvPageBar = (SDVReaderMainPagebar *)mainPagebar;
-        [sdvPageBar showSwipeForArticleViewLabel:[self isCurrentPageArticle]];
+
+        if ([self isCurrentPageArticle]) {
+            [self showReaderViewSwipe];
+        } else {
+            if (readerViewTimer != nil) {
+                [readerViewTimer invalidate];
+                readerViewTimer = nil;
+            }
+            swipeForArticleView.hidden = YES;
+        }
     }
 }
 
+-(void)hideReaderViewSwipe
+{
+    [UIView animateWithDuration:1.0 delay:0.0
+        options:UIViewAnimationOptionCurveLinear | UIViewAnimationOptionAllowUserInteraction
+                     animations:^(void)
+        {
+            swipeForArticleView.alpha = 0.0f;
+        }
+        completion:^(BOOL finished)
+        {
+            swipeForArticleView.hidden = YES;
+            swipeForArticleView.alpha = 1.0f;
+        }
+     ];
+    
+    if (readerViewTimer != nil) {
+        [readerViewTimer invalidate];
+        readerViewTimer = nil;
+    }
+}
+
+-(void)showReaderViewSwipe
+{
+    swipeForArticleView.alpha = 1.0f;
+    swipeForArticleView.hidden = NO;
+    
+    if (readerViewTimer != nil) {
```

Comment:
Can you put this check in a separate method, since you are calling thrice in the same class?